distributed/scheduler.py

```diff
@@ -1336,6 +1337,8 @@ def heartbeat_worker(
         host_info = host_info or {}
 
         self.host_info[host]["last-seen"] = local_now
+        frac = 1 / 20 / len(self.workers)
```

Comment:
This is a bit over my head: Why do we need to divide by the number of workers?

On a large cluster with little activity, I fear that this could lead to very small convergence.

Reply:
Each worker sends a heartbeat independently.  So the more workers there are, the more heartbeats there are.  This delay time means (I think) that our bandwidth value changes with a time scale of around 20 heartbeats.  This is probably about a minute of time, so should be fairly fast on the scheduler side.  The worker's heartbeat also has a time scale though, and it only changes when there are actual communications, so that might move more slowly.